dimscord/helpers.nim

```diff
@@ -520,5 +520,21 @@ proc waitForComponentUse*(client: DiscordClient, id: string): Future[Interaction
     i.data.unsafeGet().interactionType == idtMessageComponent and
     i.data.get().custom_id == id
 
+proc orTimeout*[T](fut: Future[T], time: TimeInterval): Future[Option[T]] {.async.} =
+  ## Helper that returns `none(T)` if a Future timeouts.
+  ## Returns `some(T)` is it finished within time limit
+  # We need time in milliseconds
+  let milliseconds = initDuration(
+    time.nanoseconds,
+    time.microseconds,
+    time.milliseconds,
+    time.seconds,
+    time.minutes,
+    time.hours,
+    time.days,
+    time.weeks).inMilliseconds()
+
+  if await fut.withTimeout(milliseconds):
+    result = some await fut
 
 discard DiscordClient().waitForReply(Message())
```

Comment:
whats the line for?

Reply:
That's for me testing and forgetting to remove the line 😛